### PR TITLE
map only one row - mapRowFields

### PR DESCRIPTION
**Pull request by guilemos**
Support to map only one row that lives at a array/object when we are out of a loop.

This is useful when working with "active row".
```javascript
const state = {
  byId: {
    "/rooms/1": {
        name: "Nice room"
    },
    "/rooms/2": {
        name: "Nice room 2"
    },
  },
  allIds: ["/rooms/1", "/rooms/2"],
};
```
```javascript
...mapRowFields('rooms', { item: 'byId[/rooms/2]' }),
```
```javascript
<template>
    <v-input v-model="item.name" />
</template>
```
However, I am stuck with the best way to use it. For this PR be worth, we should be able to make row parameter dynamic.
```javascript
...mapRowFields('rooms', { item: 'byId[/rooms/2]' }), //<-- '/rooms/2' must be dynamic
```
Reading the docs, I think same happens here: `'addresses[0].town',`
```javascript
export default {
  computed: {
    // When using nested data structures, the string
    // after the last dot (e.g. `firstName`) is used
    // for defining the name of the computed property.
    ...mapFields([
      'user.firstName',
      'user.lastName',
      // It's also possible to access
      // nested properties in arrays.
      'addresses[0].town',
    ]),
  },
};
```
I think this is a limitation related with [vuex#863](https://github.com/vuejs/vuex/issues/863#issuecomment-329510765)
Maybe I can make this PR better using a 'active property' approach, like:
```javascript
const state = {
  byId: {
    "/rooms/1": {
        name: "Nice room"
    },
    "/rooms/2": {
        name: "Nice room 2",
        vuexMapFieldsActiveRow: true
    },
  },
  allIds: ["/rooms/1", "/rooms/2"],
};
```
```javascript
...mapRowFields('rooms', { item: 'byId' }),
```
```javascript
<template>
    <v-input v-model="item.name" /> //<-- Two-wat enabled for row '/rooms/2' 
</template>
```
Any ideas?

Best regards

### CODE REVIEW

**Comment by guilemos** [comment]
I think this PR solves #43

**Comment by guilemos** [comment]
Hello @maoberlehner, could you take a look at this? Thank you

**Comment by guilemos** [comment]
I move forward with this PR in my project and solve the 'dynamic parameter' like this:

```javascript

  data() {
    return {
      itemId: this.$route.params.room ? decodeURIComponent(this.$route.params.room) : uuid(),
    };
  },
```
 
```javascript
 ...mapRowFields('rooms', { item: (c) => { return `byId[${c.itemId}]`; } }),
```

**Comment by maoberlehner** [comment]
Hey @guilemos thanks for contributing!

There are a couple of things that need to be fixed before I can consider to merge this PR:

- It looks like you changed permissions on a couple (all?) files. In the "Files changed" tab are a lot of files listed with no changes. Please fix that.
- You unnecessarily removed whitespace [here](https://github.com/maoberlehner/vuex-map-fields/pull/88/files#diff-1fdf421c05c1140f6d71444ea2b27638L53) and [here](https://github.com/maoberlehner/vuex-map-fields/pull/88/files#diff-1fdf421c05c1140f6d71444ea2b27638L71).
- Please install and use [ESLint](https://eslint.org/) and fix all the linting errors.
- Please update the README with information how and when to use this. Most importantly focus on what's the difference to `mapMultiRowFields()` and when to use one over the other.

Thanks!

**Comment by guilemos** [comment]
Hi @maoberlehner,

Once mapRowFields feature is much more similar to mapFields than mapMultiRowFields, I wondering if make more sense add two features to mapFields function:

1. Ability to map all properties when path return a object, e.g.: `users[0]`;
2. Ability to mapFields to accept functions as path to make feature above dynamic.

So I did submit PR #95 which implements the solution above. I'll keep PR #88 so you can decide which one is a best solution.

In short, PR #88 creates a new function semantically similar to mapFields once target is "map fields" of only one "row" inside a array,  just like current mapFields function does inside "store". #95 add 2 new abilities to the mapFields function so we can avoid 2 functions with same semantic.

Thank you!

**Comment by guilemos** [comment]
I’ll update the docs after you decide.

Regards
